redgenes/utils.py

```diff
@@ -1,29 +1,138 @@
+import gzip
+import shutil
 import subprocess
+from BCBio import GFF
+from pathlib import Path
+from contextlib import contextmanager
+from redgenes.exceptions import InvalidFna
 
 
-def run_command(commands, errortext):
-    """Basic function to run bash commands using subprocess."""
+################################
+# Generate sql insert statements
+################################
+def generate_insert_stmt(table_name, column_names):
+    stmt = f"insert into {table_name} ({', '.join(column_names)}) values ({', '.join(['?']*len(column_names))});"
+    return stmt
+
+
+################################
+# Extract data from GFF3 files
+################################
+def read_gff_file(gff_file):
+    """Parse a gff file and return all of its records"""
+    if not Path(gff_file).exists():
+        raise FileNotFoundError(f"Error: GFF file not found at {gff_file}")
+    records = []
+    with open(gff_file, "r") as in_handle:
+        records = list(GFF.parse(in_handle))
```

Comment:
I think @rolesucsd shifted to `skbio` for `GFF` parsing? Note that we are involved in skbio development and doing what we can to use and make that library better is a big big win